picamera2/picamera2.py

```diff
@@ -1116,6 +1116,15 @@ def configure_(self, camera_config="preview") -> None:
         self.controls = Controls(self, controls=self.camera_config['controls'])
         self.configure_count += 1
 
+        if "ScalerCrops" in self.camera_controls:
+            par_crop = self.camera_controls["ScalerCrops"]
```

Comment:
Ah, I'm guessing `par_crop` means `preserve_ar_crop`? Actually, for my curiosity, how does this work? Presumably libcamera is calculating the AR-preserving crop for each of the cameras - except I thought asking for camera controls was supposed to give us min/max/default triple. So I think I'm a bit confused!

Reply:
> Presumably libcamera is calculating the AR-preserving crop for each of the cameras - except I thought asking for camera controls was supposed to give us min/max/default triple. So I think I'm a bit confused!

This is a bit of a hack where the controlinfo for rpi::ScalerCrops actually gives us the default (i.e. preserved ar) crop for output 0 and output 1 in the min/max fields.  Unfortunately there is no other mechanism that can be used to get this out for span controls.

Reply:
I see, yes, that's a bit sneaky! Maybe worth a comment as it was a bit head-scratchy?